### PR TITLE
gNOI-3.3: add new subtests for configuration convergence

### PR DESCRIPTION
**Pull request by kjahed**
Add the following subtests:
1- TestConfigurationUnchanged checks that the configuration is unchanged after switchover
2- TestConfigurationConvergence checks that a large configuration is accepted within 60 seconds after switchover

Also adds a [Reporter](https://pkg.go.dev/github.com/google/go-cmp/cmp#Reporter) to collect diff entries from cmp.Diff. This allows the test to examine the each diff to decide whether, for e.g., the configuration match. In this enhanced switchover test, this is used to ignore leaves that are not explicitly configured (i.e., fields with nil value) when comparing against the configuration returned from the device.

### CODE REVIEW

**Comment by LimeHat** [comment]
Can you please add 
* `new-requirement-pr` tag to the PR
* an updated README that introduces these requirements?

Thanks,

**Comment by LimeHat** [comment]
Thanks for the updates @kjahed 

I'm wondering if these subtests belong to gNOI-3.3?

Specifically for the second case, configuration convergence, there's a separate test gNMI-1.19
https://github.com/openconfig/featureprofiles/tree/main/feature/platform/controllercard/tests/SetRequest_controll_card_switchover

cc @rszarecki  @dplore  @cfernanz

File: feature/gnoi/system/tests/supervisor_switchover_test/README.md [hunk 1]
```diff
@@ -6,12 +6,23 @@ Validate that the active supervisor can be switched.
...
-    the one specified in the request.
-*   Validate the standby RE/SUP becomes the active after switchover
-*   Validate that all connected ports are re-enabled.
+*   Basic Switchover
+    *   Issue gnoi.SwitchControlProcessor to the chassis with dual supervisor,
+        specifying the path to choose the standby RE/SUP.
+    *   Ensure the SwitchControlProcessorResponse has the new active supervisor as
+        the one specified in the request.
+    *   Validate the standby RE/SUP becomes the active after switchover
+    *   Validate that all connected ports are re-enabled.
```
Comment (dplore) [inline feature/gnoi/system/tests/supervisor_switchover_test/README.md:15]:
```suggestion
*  gNOI-3.3.1 Basic Switchover
    *   Issue gnoi.SwitchControlProcessor to the chassis with dual `CONTROLLER_CARD`,
        specifying the path to choose the standby RE/SUP.
    *   Ensure the SwitchControlProcessorResponse has the new active `CONTROLLER_CARD` as
        the one specified in the request.
    *   Validate the standby `CONTROLLER_CARD` becomes the active after switchover
    *   Validate that all connected ports are re-enabled.
```

File: feature/gnoi/system/tests/supervisor_switchover_test/README.md [hunk 2]
```diff
@@ -6,12 +6,23 @@ Validate that the active supervisor can be switched.
...
+
+*   Configuration consistent after switchover
+    *   Get the configuration before switchover.
+    *   Perform and validate switchover using the process outlined above.
+    *   Get configuration after switchover and validate that it matches the pre-switchover configuration.
+
+*  Configuration convergence after switchover
+    *   Perform and validate switchover using the process outlined above.
+    *   Push a large configuration
+    *   Get the configuration and validate that it is accepted whithin 60 seconds.
```
Comment (dplore) [inline feature/gnoi/system/tests/supervisor_switchover_test/README.md:25]: I think these steps should be moved to gNMI-1.19?

I think this portion of the test is duplicate with gnmi-1.19?   The implementation should probably be moved here instead:

https://github.com/openconfig/featureprofiles/blob/64388db0ea85c18ed6e556324dc8f2dcc643ddfb/feature/platform/controllercard/tests/SetRequest_controll_card_switchover/README.md?plain=1#L1

**Review by dplore — COMMENTED** [review]
The implementation should probably be moved here instead:

https://github.com/openconfig/featureprofiles/blob/64388db0ea85c18ed6e556324dc8f2dcc643ddfb/feature/platform/controllercard/tests/SetRequest_controll_card_switchover/README.md?plain=1#L1

File: feature/gnoi/system/tests/supervisor_switchover_test/README.md (hunk 2 above)
Comment (kjahed) [reply feature/gnoi/system/tests/supervisor_switchover_test/README.md:25]: Ack. Will close this and open a new PR for gNMI-1.19.

**Pull request closed without merging**